File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        mp = {}
        l = 0
        ans = 0
        for r in range(len(s)):
            if s[r] in mp:
                l = max(mp[s[r]]+1, l)
            mp[s[r]] = r
            ans = max(ans, r - l + 1)
        return ans
        

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    // int dfs(TreeNode* root, int &mx) {
    //     if(root == nullptr){
    //         return 0;
    //     }
    //     mx = max(mx, 1+dfs(root->left, mx));
    //     mx = max(mx, 1+dfs(root->right, mx));
    //     return mx;
    // }
    int maxDepth(TreeNode* root) {
        if(root == nullptr) {
            return 0;
        }
        int ans = 0;
        ans = max(ans, 1+maxDepth(root->left));
        ans = max(ans, 1 + maxDepth(root->right));
        return ans;
    }
};
